docs/extensibility-http.md

```diff
@@ -0,0 +1,483 @@
+  --target-port=8080 \
+  --labels=project=akri,broker=http,function=device
+done
+```
+
+> Optional: check one the services:
+>
+> ```bash
+> kubectl run curl \
+> --stdin --tty --rm \
+> --image=radial/busyboxplus:curl
+> ```
+>
+> Then, pick a value for `X` between 1 and 9:
+>
+> ```bash
+> X=6
+> curl device-${X}:8080
+> curl device-${X}:8080/
+> curl http://device-${X}:8080/
+> curl http://device-${X}.default:8080/
+> ```
+>
+> Any or all of these should return a (random) 'sensor' value.
+
+Then apply `discovery.yaml` to create a Deployment (called `discovery`) and a Pod (called `discovery-...`):
+
+```bash
+kubectl apply --filename=./discovery.yaml
+```
+
+Then create a Service (called `discovery`) using the deployment:
+
+```bash
+kubectl expose deployment/discovery \
+--name=discovery \
+--port=9999 \
+--target-port=9999 \
+--labels=project=akri,broker=http,function=discovery
+```
+
+> Optional: check the service to confirm that it reports a list of devices correctly using:
+> 
+> ```bash
+> kubectl run curl \
+> --rm --stdin --tty \
+> --image=radial/busyboxplus:curl
+> ```
+>
+> Then, curl the service's endpoint:
+>
+> ```bash
+> curl discovery:9999/
+> ```
+>
+> This should return a list of 9 devices, of the form `http://device-X:8080`
+
+## Revise Helm Deployment
+
+For the agent to correctly resolve in-cluster DNS addresses and use host networking, we need to add `dnsPolicy: ClusterFirstWithHostNet` in the agent's spec:
+
+```bash
+sed \
+--in-place \
+'s|\s\{6\}hostNetwork: true|      hostNetwork: true\n      dnsPolicy: ClusterFirstWithHostNet|g' \
+./akri/deployment/helm/templates/agent.yaml
+```
+
+## Deploy Akri
+
+> Optional: If you've previous installed Akri and wish to reset, you may:
+>
+> ```bash
+> # Delete Akri Helm
+> sudo microk8s.helm3 uninstall akri
+>
+> # Delete Akri CRDs
+> kubectl delete crd/configurations.akri.sh
+> kubectl delete crd/instances.akri.sh
+> ```
+
+Deploy the revised (!) Helm Chart to your cluster:
+
+```bash
+HOST="ghcr.io"
+USER="..."
+REPO="${HOST}/${USER}"
+
+sudo microk8s.helm3 install akri ./akri/deployment/helm \
+--set imagePullSecrets[0].name="${HOST}" \
+--set agent.image.repository="${REPO}/agent" \
+--set agent.image.tag="v0.0.38-amd64" \
+--set controller.image.repository="${REPO}/controller" \
+--set controller.image.tag="v0.0.38-amd64"
+```
+
+> **NOTE** the Akri version (`v0.0.38`) may change
+
+Check using `kubectl get pods` and look for a pod named `akri-agent-...` and another named `akri-controller...` and that they're both `RUNNING`.
+
+Alternatively, you may:
+
+```bash
+kubectl get pods --selector=name=akri-agent
+kubectl get pods --selector=app=akri-controller
+```
+
+Also:
+
+```bash
+kubectl get crds --output=name
+customresourcedefinition.apiextensions.k8s.io/configurations.akri.sh
+customresourcedefinition.apiextensions.k8s.io/instances.akri.sh
+```
+
+## Deploy standalone Broker
+
+Now we can deploy the (standlone) broker.
+
+Ensure the `image` value is correct.
```

Comment:
can we add a sed command using the env variables you've established to ensure `image` is correct?

Reply:
+ Made the use of the container images consistent throughout the document.
+ Added a (MIT) `LICENSE` to the device|discovery repo so that it matches Akri's `LICENSE`
+ Revised the Kubernetes config to make it more generic and easier to `sed`

Fixed: https://github.com/deislabs/akri/pull/135/commits/481e37589514a69cd3874331dcf3e7b087777fa6